src/main/java/net/logstash/logback/appender/AsyncDisruptorAppender.java

```diff
@@ -422,57 +450,127 @@ public void stop() {
         if (!super.isStarted()) {
             return;
         }
+        
         /*
          * Don't allow any more events to be appended.
          */
         super.stop();
-        try {
-            this.disruptor.shutdown(1, TimeUnit.MINUTES);
-        } catch (TimeoutException e) {
+
+        
+        /*
+         * Shutdown Disruptor
+         *
+         * Calling Disruptor#shutdown() will wait until all enqueued events are fully processed,
+         * but this waiting happens in a busy-spin. To avoid wasting CPU we wait for at most the configured
+         * grace period before asking the Disruptor for an immediate shutdown.
+         */
+        long deadline = getShutdownGracePeriod().getMilliseconds() < 0 ? Long.MAX_VALUE : System.currentTimeMillis() + getShutdownGracePeriod().getMilliseconds();
+        while (!isRingBufferEmpty() && (System.currentTimeMillis() < deadline)) {
+            LockSupport.parkNanos(SLEEP_TIME_DURING_SHUTDOWN);
+        }
+        
+        this.disruptor.halt();
+        
+        
+        if (!isRingBufferEmpty()) {
             addWarn("Some queued events have not been logged due to requested shutdown");
         }
 
+        
+        /*
+         * Shutdown executor service
+         */
         this.executorService.shutdown();
 
         try {
-            if (!this.executorService.awaitTermination(1, TimeUnit.MINUTES)) {
-                addWarn("Some queued events have not been logged due to requested shutdown");
-            }
```

Comment:
why remove the warning?

Reply:
Because the executorService isn't actually processing any event, only housekeeping work. Events are processed on threads scheduled by the disruptor. Therefore I thought it was enough to issue a warning only if some events remained in the buffer at the end of the shutdown procedure.

Reply:
Got it.  Thanks